src/main/java/org/chocosolver/solver/search/IResolutionHelper.java

```diff
@@ -0,0 +1,384 @@
+ *    This product includes software developed by the <organization>.
+ * 4. Neither the name of the <organization> nor the
+ *    names of its contributors may be used to endorse or promote products
+ *    derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ''AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.chocosolver.solver.search;
+
+import java.util.List;
+import java.util.Spliterator;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.chocosolver.solver.ISelf;
+import org.chocosolver.solver.Model;
+import org.chocosolver.solver.ResolutionPolicy;
+import org.chocosolver.solver.Solution;
+import org.chocosolver.solver.objective.ParetoOptimizer;
+import org.chocosolver.solver.search.limits.ACounter;
+import org.chocosolver.solver.search.measure.IMeasures;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.util.ESat;
+import org.chocosolver.util.criteria.Criterion;
+
+/**
+ * Interface to define most commonly used resolution procedures.
+ * <p>
+ * Project: choco-solver.
+ *
+ * @author Jean-Guillaum Fages
+ * @author Charles Prud'homme
+ * @since 25/04/2016.
+ */
+public interface IResolutionHelper extends ISelf<Model> {
+
+	/**
+	 * Attempts to find a solution of the declared problem.
+	 * <ul>
+	 * <li>If the method returns <tt>null</tt>:</li>
+	 * <ul>
+	 * <li>either a stop criterion (e.g., a time limit) stops the search before a solution has been found,</li>
+	 * <li>or no solution exists for the problem (i.e., over-constrained).</li>
+	 * </ul>
+	 * <li>if the method returns a {@link Solution}:</li>
+	 * <ul>
+	 * <li>a solution has been found. This method can be called anew to look for the next solution, if any.</li>
+	 * </ul>
+	 * </ul>
+	 * <p>
+	 * If a solution has been found, since the search process stops on that solution, variables' value can be read, e.g.,
+	 * {@code intvar.getValue()} or the solution can be recorded:
+	 *
+	 * <pre>
+	 * {
+	 * 	&#64;code
+	 * 	Solution s = new Solution(model);
+	 * 	s.record();
+	 * }
+	 * </pre>
+	 * <p>
+	 * This method run the following instructions:
+	 *
+	 * <pre>
+	 *     {@code
```

Comment:
The preview code should be updated with the method body, since we want this class to be a helper, so anyone can copy/paste and modify the code